internal/exec/exec.go

```diff
@@ -62,6 +64,8 @@ type fieldToExec struct {
 	sels     []selected.Selection
 	resolver reflect.Value
 	out      *bytes.Buffer
+	lock     sync.Mutex
```

Comment:
We need this mutex to prevent race conditions where we might try to write the output and resolve the field at the same time. The lock prevents us from updating the `finished` bool or writing the output until whichever finished first is completed. 


Is this correct?

Reply:
Yup!